docs(tutorial): add dart version of toh tutorial

kasperpeulen opened the pull request:
@kwalrath in this PR, I just focused on getting the Dart version up to date with the typescript version

In the other PR I ported the end result to dart, but it seems like this code was only used for plnkr, or maybe it will be used for the last part (toh-pt5), but this is not yet done for typescript it seems.

zoechi commented on public/docs/dart/latest/tutorial/dependency-injection.jade:
```diff
@@ -0,0 +1,623 @@
+    private engine: Engine;
+    private tires: Tires;
+  
+    constructor() {
```
should be `Tires` instead of `constructor`

zoechi commented on public/docs/dart/latest/tutorial/dependency-injection.jade:
````diff
@@ -0,0 +1,623 @@
+  That's super easy. We probably already know what to do. We change our `Car` constructor to this:
+  <a name="ctor-injection"></a>
+  ```
+  constructor(engine: Engine, tires: Tires) {
````
`Car`

zoechi commented on public/docs/dart/latest/tutorial/dependency-injection.jade:
````diff
@@ -0,0 +1,623 @@
+  We'll begin with a simplified version of the `HeroesComponent` 
+  that we built in the [The Tour of Heroes](../tutorial/).
+  ```
+  import {Component} from 'angular2/core';
````
TS code

zoechi commented on public/docs/dart/latest/tutorial/dependency-injection.jade:
````diff
@@ -0,0 +1,623 @@
+    Write this service in its own file. See [this note](#forward-ref) to understand why.
+:markdown
+  ```
+  import {Hero} from './hero';
````
TS code

zoechi commented on public/docs/dart/latest/tutorial/dependency-injection.jade:
````diff
@@ -0,0 +1,623 @@
+  earlier](#ctor-injection)".
+  
+  ```
+  constructor(heroService: HeroService) {
````
TS code

zoechi commented on public/docs/dart/latest/tutorial/dependency-injection.jade:
````diff
@@ -0,0 +1,623 @@
+  Mission accomplished!  We don't even need the Angular Dependency Injection system to test the `HeroesComponent`.
+  We simply create a bew `HeroesComponent` with a mock service and poke at it:
+  ```
+  it("should have heroes when created", () => {
````
TS code

kasperpeulen commented:
Oh oopz @zoechi I have only converted toh-1 up to toh-4 to dart.

I first just copied all the ts source files from the ts directory to the dart directory ,but I think the DI injection files shouldn't actually be here, so I removed them now (should probably also removed from the ts source).

Also toh-5 is not converted, because the ts version is not yet finished.

zoechi commented on public/docs/_examples/toh-4/dart/lib/app_component.dart:
```diff
@@ -0,0 +1,106 @@
+      <h2>My Heroes</h2>
+      <ul class="heroes">
+        <li *ngFor="#hero of heroes"
+          [class.selected]="hero === selectedHero"
```
`===` not valid Dart

kasperpeulen replied:
thanks, fixed

zoechi commented:
The `toh-pt5.jade` file is still included, is this intentional?

zoechi commented on public/docs/_examples/toh-2/dart/lib/app_component.dart:
```diff
@@ -0,0 +1,114 @@
+      <h2>My Heroes</h2>
+      <ul class="heroes">
+        <li *ngFor="#hero of heroes"
+          [class.selected]="hero === selectedHero"
```
TS `===`

zoechi commented on public/docs/dart/latest/tutorial/toh-pt2.jade:
```diff
@@ -0,0 +1,309 @@
+  :marked
+    Notice in the template that the `class.selected` is surrounded in square brackets (`[]`).
+    This is the syntax for a Property Binding, a binding in which data flows one way
+    from the data source (the expression `hero === selectedHero`) to a property of `class`.
```
`===`

zoechi commented on public/docs/dart/latest/tutorial/toh-pt3.jade:
```diff
@@ -0,0 +1,260 @@
+
+  We create metadata with the `@Component` decorator where we
+  specify the selector name that identifies this component's element.
+  Then we export the class to make it available to other components.
```
We don't actually export a class in Dart, at least not explicit

zoechi commented on public/docs/dart/latest/tutorial/toh-pt3.jade:
```diff
@@ -0,0 +1,260 @@
++makeExample('toh-3/dart/lib/hero.dart', null, 'hero.dart (Exported Hero class)')(format=".")
+
+:marked
+  We export the `Hero` class from `hero.dart` because we'll need to reference it in both component files.
```
We don't explicitly export in Dart

kasperpeulen commented:
@zoechi I removed that file as well now, and fixed the other comments.

zoechi commented on public/docs/_examples/toh-4/dart/lib/app_component_1.dart:
```diff
@@ -0,0 +1,64 @@
+  // #docregion getHeroes
+  getHeroes() {
+    //#docregion get-heroes
+    heroes = _heroService.getHeroes();
```
shouldn't this use `async` / `await` ?

zoechi replied:
This is ok, it uses `hero_service_1.dart` where the value is returned sync.

zoechi commented on public/docs/_examples/toh-4/dart/lib/app_component_1.dart:
```diff
@@ -0,0 +1,64 @@
+  ngOnInit() {
+    // #enddocregion on-init
+    getHeroes();
+    // #docregion on-init
```
shouldn't this use async / await ?

zoechi replied:
This is ok, it uses `hero_service_1.dart` where the value is returned sync.

zoechi commented:
I did just a shallow review but now LGTM

kwalrath commented:
Taking a very belated look now...

kwalrath commented:
@kasperpeulen could you please `git rebase upstream/master`, resolve any conflicts, and push? If that's not practical, I guess we should start a new PR.

You can rename your .jade files to *-NEW.jade to get around some of the conflicts. And just drop the _data.json changes.

kwalrath commented:
Just FYI, I'm copying down the toh-1 code to submit in a separate PR.

kwalrath commented:
Closing, now that all changes have gotten into merged PRs. Thank you, @kasperpeulen!
